source/MaterialXCore/Util.cpp

```diff
@@ -164,4 +165,15 @@ string parentNamePath(const string& namePath)
     return EMPTY_STRING;
 }
 
+ElementPtr changeChildCategory(ElementPtr parent, ElementPtr origChild, const string& category)
```

Comment:
This is code we use externally to change node categories for node translation.

Reply:
Thanks for the clarification, and that makes sense.  I may go ahead and merge this new method with the existing updateChildSubclass in Document.cpp, so that we have just one implementation of this logic in the codebase.  The merged version could potentially be a method on the Element class.